app/scripts/controllers/UserModalCtrl.js

```diff
@@ -0,0 +1,20 @@
+(function() {
+    function UserModalCtrl($scope, $uibModalInstance, $cookies) {
+        
+        $scope.setUsername = function(newUsername) {
+            if($scope.newUsername && $scope.newUsername !== ''){
+                
+                $cookies.blocChatCurrentUser = $scope.newUsername;
```

Comment:
This will not give you an actual error (thanks, Angular!) but it will not produce the expected result. $cookies is like a data store where you set a `key` and give it a `value`. You can then `get` the value out of the store by passing in the name of the key (very much like an object). To see what I mean, drop a `console.log` statement that evaluates `$cookies.get('blocChatCurrentUser')` after you set it and see that it should be undefined. Now, check out https://docs.angularjs.org/api/ngCookies/service/$cookies#put and implement it in your code and it should log out the stored username.